core/network/src/protocol/block_requests.rs

```diff
@@ -0,0 +1,336 @@
+//! `NetworkBehaviour` implementation which handles incoming block requests.
+//!
+//! Every request is coming in on a separate connection substream which gets
+//! closed after we have sent the response back. Incoming requests are encoded
+//! as protocol buffers (cf. `api.v1.proto`).
+
+use codec::{Encode, Decode};
+use crate::{
+	chain::Client,
+	protocol::{api, message::BlockAttributes}
+};
+use futures::prelude::*;
+use libp2p::{
+	core::{
+		ConnectedPoint,
+		Multiaddr,
+		PeerId,
+		upgrade::{InboundUpgrade, ReadOneError, ReadRespond, UpgradeInfo, WriteOne, Negotiated, read_respond},
+		upgrade::{DeniedUpgrade, write_one}
+	},
+	swarm::{NetworkBehaviour, NetworkBehaviourAction, OneShotHandler, PollParameters, SubstreamProtocol}
+};
+use log::{debug, trace};
+use prost::Message;
+use sr_primitives::{generic::BlockId, traits::{Block, Header, One, Zero}};
+use std::{
+	cmp::min,
+	collections::VecDeque,
+	io,
+	iter,
+	sync::Arc,
+	time::Duration
+};
+use tokio_io::{AsyncRead, AsyncWrite};
+use void::{Void, unreachable};
+
+// Type alias for convenience.
+pub type Error = Box<dyn std::error::Error + 'static>;
+
+/// Configuration options for `BlockRequests`.
+#[derive(Debug, Clone)]
+pub struct Config {
+	max_block_data_response: u32,
+	max_request_len: usize,
+	inactivity_timeout: Duration,
+	protocol: &'static [u8],
+}
+
+impl Default for Config {
+	fn default() -> Self {
+		Config::new()
+	}
+}
+
+#[allow(unused)]
+impl Config {
+	/// Create a fresh configuration with the following options:
+	///
+	/// - max. block data in response = 128
+	/// - max. request size = 1 MiB
+	/// - inactivity timeout = 15s
+	/// - protocol string = b"/polkadot/sync/1"
+	pub fn new() -> Self {
+		Config {
+			max_block_data_response: 128,
+			max_request_len: 1024 * 1024,
+			inactivity_timeout: Duration::from_secs(15),
+			protocol: b"/polkadot/sync/1",
```

Comment:
I don't think there should be a default value for that.
The value should be `format!("/{}/sync/1", protocol_id)` (where `protocol_id` is the `protocol_id` of type `ProtocolId` that is passed around during the network's initialization), and that must be passed by parameter.